### PR TITLE
fix: add a fallback for fullstack cert 

### LINKED ISSUE #18128: Full Stack Cert - Shows current date(not claimed date) - Shows as legacy cert on profile page
#### Describe your problem and - if possible - how to reproduce it
1) The Full Stack Cert claimed date will only show as the current date and not the date it was claimed on. I claimed mine on the 7th of September and checking back regularly I noticed the claimed on date was always the current date.

2) The Full Stack Cert appears as the  Legacy Full Stack Cert on a users profile page, clicking this leads to an error as the legacy cert does not exist. It should be in with the new certificates.
That being said, on the settings page the link to view the certificate is linked correctly.

#### Add a Link to the page with the problem
https://www.freecodecamp.org/certification/       ${username-here}         /full-stack
https://www.freecodecamp.org/         ${username-here}

#### Tell us about your browser and operating system
N.A

#### If possible, add a screenshot here
1)
Just the claimed on date. No screenshot necessary.

2)
issue2

### PR DESCRIPTION
**Pull request by ahmaxed**
Refers #18128

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
I can make the code more declarative if needed.

**Comment by raisedadead** [comment]
Yes, please.

**Review by moT01 — COMMENTED** [review]
I can confirm that this works locally...
Screen Shot 2019-11-13 at 10 17 12 AM

**Comment by moT01** [comment]
Bump

**Comment by raisedadead** [comment]
Hi Tom, as you probably know we are swamped a little bit. I was wondering if you would be interested in taking this up (QA and pushing updates to the PR as needed)?

**Comment by moT01** [comment]
I'm not sure what parts were all in need of being more declarative. But I changed the loops to use array/object methods. Let me know if it's not what we want, or something else needs to be done.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ahmadabdolsaheb** [comment]
@moT01, thanks for working on this. just a reminder that this bandage solution adds a fallback if we don't find a record associated with a fullcert challenge. the main issue is that we are not recording the fullcert challenge only setting the isFullStackCert to true. In order to resolve #18128, we would need to address the main issue as well.

File: api-server/server/boot/certificate.js [hunk 1]
```diff
@@ -438,18 +438,30 @@ function createShowCert(app) {
 
         // the challenge id has been rotated for isDataVisCert
         // so we need to check for id 561add10cb82ac38a17513b3
+
+        const oldDataVizId = '561add10cb82ac38a17513b3';
```
Comment (raisedadead) [inline api-server/server/boot/certificate.js]: Should we keep this in the config directory in the root?

**Comment by raisedadead** [comment]
I have also changed the PR to be non closing to make sure we track and address the parent issue.

**Comment by ojeytonwilliams** [comment]
I think I approved this a little prematurely - I made sure the code was good (it is!) but I didn't think about the broader implications.  

As for Ahmad's point about saving the challenge completion, presumably another certification needs creating, in `curriculum/challenges/english/09-certificates` for full stack.  Possibly it will need handling as a special case, I'm not 100% sure about the cert claiming logic.

**Comment by raisedadead** [comment]
> I think I approved this a little prematurely ...

No you didn't we need this anyways as a fallback right now. Thanks for the QA.

> ... presumably another certification needs creating ...

As for the logic, I think we would be considering awarding the full stack cert automatically as all other certs have been completed. But this needs more discussion internally as well. For now we need the work around that this PR is making.

**Comment by ojeytonwilliams** [comment]
Everything seemed to be working fine locally, so I went ahead and merged.

**Pull request merged**